framework/src/org/checkerframework/framework/type/AnnotatedTypeLoader.java

```diff
@@ -0,0 +1,418 @@
+package org.checkerframework.framework.type;
+
+import org.checkerframework.common.basetype.BaseTypeChecker;
+import org.checkerframework.framework.util.AnnotatedTypes;
+import org.checkerframework.framework.util.AnnotationBuilder;
+import org.checkerframework.javacutil.ErrorReporter;
+
+import java.io.File;
+import java.io.IOException;
+import java.lang.annotation.Annotation;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Target;
+import java.net.JarURLConnection;
+import java.net.URL;
+import java.util.Enumeration;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.jar.JarEntry;
+import java.util.jar.JarFile;
+
+import javax.annotation.processing.ProcessingEnvironment;
+import javax.lang.model.element.AnnotationMirror;
+
+/*>>>
+import org.checkerframework.checker.nullness.qual.Nullable;
+ */
+
+/**
+ * This class assists the {@link AnnotatedTypeFactory} by reflectively looking
+ * up the list of annotation class names in each checker's qual directory, and
+ * then loading and returning it as a set of annotation classes. It can also
+ * look up and load annotation classes from external directories that are passed
+ * as arguments to checkers that have extension capabilities such as the
+ * Subtyping Checker, Fenum Checker, and Units Checker.
+ *
+ * Only annotation classes that have the {@link Target} meta-annotation with the
+ * value of {@link ElementType#TYPE_USE} (and optionally
+ * {@link ElementType#TYPE_PARAMETER}) are loaded. If it has other
+ * {@link ElementType} values, it won't be loaded. Other annotation classes must
+ * be manually listed in a checker's annotated type factory by overriding
+ * {@link AnnotatedTypeFactory#createSupportedTypeQualifiers()}.
+ *
+ * The methods in this class are called within
+ * {@link AnnotatedTypeFactory#loadTypeAnnotationsFromQualDir(Class[])}.
```

Comment:
No.

The class exposes 3 public methods in total. getLoadedAnnotationClasses() assists createSupportedTypeQualifiers().  loadExternalAnnotationClass(), and loadExternalAnnotationClasses() respectively support -Aquals and -AqualDirs, but only in Subtyping, Fenum, and Units Checkers.

There currently isn't any other use cases or call sites for these sets of methods. Should I add some additional documentation to indicate where the other 2 methods are called? or should I remove this line of documentation to avoid needing to update in the future?